WordPress/Classes/Utility/ImageURLs/WPImageURLHelper.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+
+/// Helper class to create a WordPress URL for various download needs, like specifying image size or using Photon.
+public class WPImageURLHelper: NSObject
+{
+    struct GravatarDefaults {
+        static let scheme = RequestScheme.Secure.rawValue
+        static let host = "secure.\(gravatarURLBase)"
+        // unknownHash = md5("unknown@gravatar.com")
+        static let unknownHash = "ad516503a11cd5ca435acc9bb6523536"
+    }
+
+    enum RequestScheme: String {
+        case Insecure = "http"
+        case Secure = "https"
+    }
+
+    enum URLComponent: String {
+        case Blavatar = "blavatar"
+        case Gravatar = "avatar"
+    }
+
+    enum ImageURLQueryField: String {
```

Comment:
Thing with `enum` is (at least to the best of my knowledge)... it's used whenever you need to pick just one case from the list, which doesn't match with this specific use case. Not just that: the enum is never used as an enum per se, but instead, it's only used to retrieve it's `rawValue`

For multiple "Cases" we've got (OptionSet)[https://developer.apple.com/reference/swift/optionset], which may also not be the best match.